### DISCUSSION TITLE
Server-Side Component with two inputs possible without hack?

### DISCUSSION THREAD

**Discussion by philschonholzer** [digitallyinduced/ihp#1115] [Q&A]
I have a server-side component with two inputs inside. I need them to have two variable values to calculate a total. 

With a hack I was able to get it work: I added the input values of each input to both actions that handle changes of the inputs. Only like that the websocket returns with two createNode instructions and every value that needs to update is actually updated. Without the hack the total value is only updated after each input hat been changed by the user.

With this hack it works. "otherValue" is the value of the other input:
https://github.com/philschonholzer/co2db/commit/9b7bf4e3c8137f83ff79aa4fa4b1e5fa7aadb8df

Here is a deployment without the hack. You can see that the total (after "=") is only updated after both inputs sliders are changed:
https://mvazyimwyigdiuhmhhfrninfnhuwkqzw.ihpapp.com/ShowCo2Producer?co2ProducerId=c8d6f416-ad52-484f-9f7e-b510dac2c158

And here it is deploy with the hack:
https://co2db.ihpapp.com/ShowCo2Producer?co2ProducerId=c8d6f416-ad52-484f-9f7e-b510dac2c158

**Comment by philschonholzer** [comment]
So I got a cleaner version now that calls the same js function from both range inputs.

https://github.com/philschonholzer/co2db/blob/b908801c4ce8b1c06c00748cc6cd30b29d757c81/Web/Component/CommonConsumption.hs#L59-L99

What I don't like yet, is that I'm referencing the input elements directly from the script by id. This can cause bugs if the ids are changed or other elements have the same id in future.

**Comment by mpscholten** [comment]
Just tested the deployment with the hack and without the hack. Both behave the same to me and the value after the `=` is updated as expected. Anything I am missing?

**Comment by philschonholzer** [comment]
At me https://mvazyimwyigdiuhmhhfrninfnhuwkqzw.ihpapp.com/ShowCo2Producer?co2ProducerId=c8d6f416-ad52-484f-9f7e-b510dac2c158 is now working also! :O Can it be that I had something in the cache that could caused the problem?

**Reply by mpscholten** [reply]
Yes :) that could actually be it

**Reply by philschonholzer** [reply]
Ok, good to know. :)
